src/BugsnagUnityForReview.mm

```diff
@@ -0,0 +1,876 @@
+}
+
+void bugsnag_registerForSessionCallbacks(const void *configuration, bool (*callback)(void *session)){
+    [((__bridge BugsnagConfiguration *)configuration) addOnSessionBlock:^BOOL (BugsnagSession *session) {    
+        return callback((__bridge void *)session);
+    }];
+}
+
+void bugsnag_registerForOnSendCallbacks(const void *configuration, bool (*callback)(void *event)){
+    [((__bridge BugsnagConfiguration *)configuration) addOnSendErrorBlock:^BOOL (BugsnagEvent *event) {       
+        return callback((__bridge void *)event);
+    }];
+}
+
+void bugsnag_registerSession(char *sessionId, long startedAt, int unhandledCount, int handledCount) {
+    BugsnagSessionTracker *tracker = Bugsnag.client.sessionTracker;
+    [tracker registerExistingSession:sessionId == NULL ? nil : [NSString stringWithUTF8String:sessionId]
+                           startedAt:[NSDate dateWithTimeIntervalSince1970:(NSTimeInterval)startedAt]
+                                user:Bugsnag.user
+                        handledCount:(NSUInteger)handledCount
+                      unhandledCount:(NSUInteger)unhandledCount];
+}
+
+void bugsnag_retrieveCurrentSession(const void *session, void (*callback)(const void *instance, const char *sessionId, const char *startedAt, int handled, int unhandled)) {
+    if([Bugsnag client].sessionTracker.runningSession == NULL)
+    {
+      callback(session, NULL, NULL, 0, 0);
+      return;
+    }
+    NSDictionary * sessionDict = [[Bugsnag client].sessionTracker.runningSession toDictionary];
+    const char *sessionId = [[sessionDict objectForKey:@"id"] UTF8String];
+    const char *timeString = [[sessionDict objectForKey:@"startedAt"] UTF8String];
+    int handled = [sessionDict[@"handledCount"] integerValue];
+    int unhandled = [sessionDict[@"unhandledCount"] integerValue];
+    callback(session, sessionId, timeString, handled, unhandled);
+}
+
+void bugsnag_markLaunchCompleted() {
+  [Bugsnag markLaunchCompleted];
+}
+
+void *bugsnag_createConfiguration(char *apiKey) {
+  NSString *ns_apiKey = [NSString stringWithUTF8String: apiKey];
+  BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:ns_apiKey];
+  config.apiKey = ns_apiKey;
+  return (void*)CFBridgingRetain(config);
+}
+
+void bugsnag_setReleaseStage(const void *configuration, char *releaseStage) {
+  NSString *ns_releaseStage = releaseStage == NULL ? nil : [NSString stringWithUTF8String: releaseStage];
+  ((__bridge BugsnagConfiguration *)configuration).releaseStage = ns_releaseStage;
+}
+
+void bugsnag_setNotifyReleaseStages(const void *configuration, const char *releaseStages[], int releaseStagesCount){
+  NSMutableSet *ns_releaseStages = [NSMutableSet new];
+  for (int i = 0; i < releaseStagesCount; i++) {
+    const char *releaseStage = releaseStages[i];
+    if (releaseStage != nil) {
+      NSString *ns_releaseStage = [NSString stringWithUTF8String: releaseStage];
+      [ns_releaseStages addObject: ns_releaseStage];
+    }
+  }
```

Comment:
This conversion from array of C strings to an NSSet is repeated a few times (also in `bugsnag_setDiscardClasses` and `bugsnag_setRedactedKeys`) and would be good to extract to a reusable function.

Reply:
Done 👍